### PR TITLE
Add query string & body to `Net::HTTP` breadcrumb

### PR DESCRIPTION
**Pull request by ojab**
## Description
They are useful, let's add them.
This is on top of https://github.com/getsentry/sentry-ruby/pull/1637 because they're touching the same code and probably #1637 would be merged faster.

### CODE REVIEW

**Comment by st0012** [comment]
@ojab hey sorry for the trouble, but do you think you can rebase the PR? thx!

**Comment by ojab** [comment]
Sure thing, rebased, CI is green.

File: sentry-ruby/lib/sentry/net/http.rb [hunk 1]
```diff
@@ -93,8 +94,12 @@ def from_sentry_sdk?
 
       def extract_request_info(req)
         uri = req.uri || URI.parse("#{use_ssl? ? 'https' : 'http'}://#{address}#{req.path}")
-        url = "#{uri.scheme}://#{uri.host}#{uri.path}" rescue uri.to_s
+        url = "#{uri.scheme}://#{uri.host}#{uri.path}"
+        url = "#{url}?#{uri.query}" if uri.query && Sentry.configuration.send_default_pii
+
         { method: req.method, url: url }
+      rescue
+        { method: req.method, url: url.to_s }
```
Comment (st0012) [inline sentry-ruby/lib/sentry/net/http.rb:102]: this should be `uri.to_s`. I'll change it after merging.

File: sentry-ruby/lib/sentry/net/http.rb [hunk 2]
```diff
@@ -50,14 +50,15 @@ def record_sentry_breadcrumb(req, res)
         return if from_sentry_sdk?
 
         request_info = extract_request_info(req)
+        request_info[:body] = req.body if Sentry.configuration.send_default_pii
```
Comment (st0012) [inline sentry-ruby/lib/sentry/net/http.rb:53]: this extraction should happen inside `extract_request_info`.

**Review by st0012 — COMMENTED** [review]
Some small comments but nothing blocking. Thanks for the awesome feature 👍

**Pull request merged**